src/utils/metadata/parser.rs

```diff
@@ -48,13 +48,14 @@ impl ParserMetadata {
     }
 
     /// Pushes a new scope to the stack
-    pub fn push_scope(&mut self) {
-        self.context.scopes.push(ScopeUnit::new())
-    }
-
-    /// Pops the last scope from the stack
-    pub fn pop_scope(&mut self) -> Option<ScopeUnit> {
-        self.context.scopes.pop()
+    pub fn push_scope<B>(&mut self, mut body: B) -> SyntaxResult
```

Comment:
Behaviour is different, so maybe a new name? Something like `parse_in_scope`?

Reply:
Renamed as `with_push_scope()` for consistency with other `with_foo()` functions.